src/Raven.Server/Documents/Sharding/Operations/ShardedQueryOperation.cs

```diff
@@ -161,21 +164,27 @@ public ShardedQueryResult CombineResults(Dictionary<int, ShardExecutionResult<Qu
         // all the results from each command are already ordered
         using (var mergedEnumerator = new MergedEnumerator<BlittableJsonReaderObject>(_sortingComparer))
         {
-            foreach (var cmdResult in results.Values)
+            foreach (var (shardNumber, cmdResult) in results)
             {
-                mergedEnumerator.AddEnumerator(GetEnumerator(cmdResult.Result.Results.Clone(_context)));
+                mergedEnumerator.AddEnumerator(GetEnumerator(cmdResult.Result.Results.Clone(_context), shardNumber));
             }
 
             while (mergedEnumerator.MoveNext())
             {
                 result.Results.Add(mergedEnumerator.Current);
             }
 
-            static IEnumerator<BlittableJsonReaderObject> GetEnumerator(BlittableJsonReaderArray array)
+            IEnumerator<BlittableJsonReaderObject> GetEnumerator(BlittableJsonReaderArray array, int shardNumber)
             {
                 foreach (BlittableJsonReaderObject item in array)
                 {
-                    yield return item;
+                    if (Debug == false)
```

Comment:
`Debug`? That is from the studio, no? Let's use it as such.

Reply:
done